fsio.go

```diff
@@ -114,7 +121,18 @@ func (fs *FileSystem) Open(p Path, offset, length int64, buffSize int) (io.ReadC
 	}
 
 	req, _ := http.NewRequest("GET", u.String(), nil)
-	rsp, err := fs.client.Do(req)
+
+	var rsp *http.Response
+	for retries := fs.Config.MaxRetries; retries > 0; retries-- {
+		// don't scope err in this loop
+		rsp, err = fs.client.Do(req)
+		if err == nil {
+			break
+		}
+		if t, ok := err.(isTemp); !ok || !t.Temporary() {
+			break
+		}
+	}
```

Comment:
Looking at the code again, I think it would be best to leave out backoff logic, at least for this PR.  My reasoning is something like this:

Backoff logic is most useful when it's multiplicative or exponential but that requires quite a bit of additional logic.
If we're not using multiplicative or exponential backoff, then it's just an additional timeout and therefore should be left out.

What are your thoughts?